domain/crossmodelrelation/state/model/remoteapplication.go

```diff
@@ -1540,3 +1568,61 @@ AND    e2.endpoint_name    = $endpointIdentifier2.endpoint_name
 
 	return relationsSuspended[0].IsSuspended, errors.Capture(err)
 }
+
+func (st *State) isRelationCrossModel(ctx context.Context, tx *sqlair.TX, key corerelation.Key) (bool, error) {
+	// Get the relation UUID from the key (2-endpoint relation).
+	// The key contains endpoint identifiers (application names and endpoint names).
+	endpoints := key.EndpointIdentifiers()
+
+	// Use the existing method to get the relation UUID.
+	// This reuses logic already present in relation_network.go.
+	relationUUID, err := st.getRegularRelationUUIDByEndpointIdentifiers(ctx, tx, endpoints[0], endpoints[1])
+	if err != nil {
+		if errors.Is(err, relationerrors.RelationNotFound) {
+			return false, errors.Errorf("relation not found for key %q: %w", key.String(), relationerrors.RelationNotFound)
+		}
+		return false, errors.Errorf("getting relation UUID: %w", err)
+	}
+
+	// Check if any application in this relation is remote.
+	isRemote, err := st.isApplicationRemote(ctx, tx, relationUUID)
+	if err != nil {
+		return false, errors.Errorf("checking if application is remote: %w", err)
+	}
+
+	return isRemote, nil
+}
+
+func (st *State) isApplicationRemote(ctx context.Context, tx *sqlair.TX, relationUUID string) (bool, error) {
+	type relationUUIDArg struct {
+		UUID string `db:"relation_uuid"`
+	}
+
+	relUUID := relationUUIDArg{UUID: relationUUID}
+
+	// Check if any application in this relation is in
+	// application_remote_offerer or application_remote_consumer.
+	stmt, err := st.Prepare(`
+	SELECT COUNT(*) AS &countResult.count
+	FROM relation_endpoint AS re
+	JOIN application_endpoint AS ae ON re.endpoint_uuid = ae.uuid
+	WHERE re.relation_uuid = $relationUUIDArg.relation_uuid
+	  AND (
+	    ae.application_uuid IN (
+	        SELECT application_uuid FROM application_remote_offerer)
+	    OR ae.application_uuid IN (
+	        SELECT consumer_application_uuid FROM application_remote_consumer)
+	  )
+	`, countResult{}, relationUUIDArg{})
```

Comment:
It might be easier to check the charm sources
https://github.com/jack-w-shaw/juju/blob/94f3c4b63ee7471468df14ed2d0ee47742896efb/domain/removal/state/model/relation.go#L43-L83